backend/src/apiserver/storage/run_store.go

```diff
@@ -365,19 +365,26 @@ func (s *RunStore) UpdateRun(runID string, condition string, workflowRuntimeMani
 	// new in the status of an Argo manifest. This means we need to keep track
 	// manually here on what the previously updated state of the run is, to ensure
 	// we do not add duplicate metadata. Hence the locking below.
-	row := tx.QueryRow("SELECT WorkflowRuntimeManifest FROM run_details WHERE UUID = ? FOR UPDATE", runID)
+	var query string
+	switch x := s.db.SQLDialect.(type) {
+	case MySQLDialect:
+		query = "SELECT WorkflowRuntimeManifest FROM run_details WHERE UUID = ? FOR UPDATE"
+	case SQLiteDialect:
+		query = "SELECT WorkflowRuntimeManifest FROM run_details WHERE UUID = ?"
+	default:
+		glog.Fatal("Unsupported SQL dialect: %v", x)
+	}
+	row := tx.QueryRow(query, runID)
 	var storedManifest string
 	if err := row.Scan(&storedManifest); err != nil {
 		tx.Rollback()
-		return util.NewInternalServerError(err, "failed to find row with run id %q", runID)
+		return util.NewInvalidInputError("Failed to update run %s. Row not found.", runID)
 	}
 
-	if s.metadataStore != nil {
```

Comment:
Could we use the null design pattern here so that we don't forget to perform this check? (i.e. we have an interface with an actual implementation calling MySQL, and actual implementation calling SQLLite, and a "NullImplementation" that does not do anything.

This pattern really helps keep the code concise and easy to read by leveraging OO instead of requiring if/then/switch statement in several places.

Reply:
I removed the comparison to nil because I now expect s.metadataStore to always be non-nil, unlike before, when it was nil in tests. 

Go is not an OO language, and I don't think it's a good practice to shoehorn OO design patterns in Go code.